### ISSUE TITLE
bad performance

### ISSUE THREAD

**Issue by sunnyForIOS** [datamllab/rlcard#32]
########## Evaluation ##########
Timestep: 629402 Average reward is 0.458

########## Evaluation ##########
Timestep: 1258220 Average reward is 0.46

########## Evaluation ##########
Timestep: 1888626 Average reward is 0.514

########## Evaluation ##########
Timestep: 2516620 Average reward is 0.506

########## Evaluation ##########
Timestep: 3144764 Average reward is 0.492

########## Evaluation ##########
Timestep: 3774566 Average reward is 0.468

########## Evaluation ##########
Timestep: 4402996 Average reward is 0.422

**this is my doudizhu_nfsp_result log**, the more trained , the worse resullt, why?

**Comment by daochenzha** [comment]
@sunnyForIOS NFSP is not originally designed for large games like Dou Dizhu. Thus, it is normal if it can not achieve good results.

We didn't tune the model since the main goal of the project is to provide an easy-to-use testbed of card games. If you want to improve the performance, you may try: (1) tune the hyperparameters to see if it can deliver better results; (2) train DQN against rule-based agent; (3) refer to recent papers of self-play for Dou Dizhu, e.g., https://www.ijcai.org/proceedings/2019/0176.pdf

**Comment by gogov5** [comment]
Can MSTC based agent be used in Rocard ？ I see a MSTC based solution is used in the paper and got a better performance. is there plan to add a MSTC based agent demo in RLCard?

**Comment by daochenzha** [comment]
@sunnyForIOS Yes, it is possible to implement MSTC. However, to implement the paper may need lots of efforts. Currently, we plan to first develop rule-based AI for Dou Dizhu, which would be easier and also helpful. We may implement this paper in the future. Contributions are welcomed if you are interested in implementing this paper.
